packages/paste-core/components/date-picker/src/DatePicker.tsx

```diff
@@ -0,0 +1,13 @@
+import * as React from 'react';
+import type {InputProps} from '@twilio-paste/input';
+import {Input} from '@twilio-paste/input';
+
+export type DatePickerProps = Omit<InputProps, 'type'>;
+
+const DatePicker = React.forwardRef<HTMLInputElement, DatePickerProps>((props, ref) => {
+  return <Input type="date" {...props} ref={ref} />;
```

Comment:
Let's move `{...props}` to before the `type="date"` to prevent overrides in JS-land.